### PR TITLE
Refactor/lombok extensions method

### PR DESCRIPTION
**Pull request by azdrojowa123**
It was hard to decide in which place @ExtensionMethod should be used. For GitMacheteBundle it was clear because format(getString(..))is not readable and used in many places. Do you think that e.q.  places like in the picture below @ExtensionMethod should also be used ? (`invokeLaterIfNeeded`)
Screenshot 2022-05-30 at 13 03 00

### CODE REVIEW

**Comment by PawelLipski** [comment]
>  Do you think that e.q. places like in the picture below @ExtensionMethod should also be used ? (invokeLaterIfNeeded)

**e.g.**! 
aeb3d8fe02932920be301c66b48f703d_gif

**Comment by PawelLipski** [comment]
> Do you think that e.q. places like in the picture below @ExtensionMethod should also be used ? (invokeLaterIfNeeded)

But no! for `invokeLaterIfNeeded` I wouldn't use extension method. It'd IIUC need to look like `ModalityState.NON_MODAL.invokeLaterIfNeeded(...)`... which is super weird, after all we don't logically call a method **on** `NON_MODAL` constant.

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchAction.java [hunk 1]
```diff
@@ -91,9 +93,9 @@ public void actionPerformed(AnActionEvent anActionEvent) {
 
       boolean isUpToDate = FetchUpToDateTimeoutStatus.isUpToDate(gitRepository);
       String fetchNotificationPrefix = isUpToDate
-          ? format(getString("action.GitMachete.BasePullBranchAction.notification.title.no-fetch-perform"),
-              FETCH_ALL_UP_TO_DATE_TIMEOUT_AS_STRING)
-          : format(getString("action.GitMachete.BasePullBranchAction.notification.title.fetch-perform"));
+          ? getString("action.GitMachete.BasePullBranchAction.notification.title.no-fetch-perform")
+              .format(FETCH_ALL_UP_TO_DATE_TIMEOUT_AS_STRING)
+          : getString("action.GitMachete.BasePullBranchAction.notification.title.fetch-perform").format();
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchAction.java:98]: Huh `format` with no params makes no sense, unless I'm missing sth 🤔

Actually, pls open an issue to add an ArchUnit test to capture such nullary `format` calls

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreCommit.java [hunk 2]
```diff
@@ -22,7 +24,7 @@ public GitCoreCommit(@NonLeaked RevCommit commit) {
     this.shortMessage = commit.getShortMessage();
     this.commitTime = Instant.ofEpochSecond(commit.getCommitTime());
     this.hash = GitCoreCommitHash.of(commit.getId());
-    this.treeHash = GitCoreTreeHash.of(commit.getTree().getId());
```
Comment (PawelLipski) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreCommit.java:25]: Huh I'd leave as is, `.of()` looks confusing as hell

Comment (PawelLipski) [reply gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreCommit.java:25]: Although `commit.getTree().getId().toGitCoreTreeHash()` would be quite reasonable

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreReflogEntry.java [hunk 3]
```diff
@@ -35,13 +37,13 @@ public Instant getTimestamp() {
   @Override
   @ToString.Include(name = "oldCommitHash")
   public Option<IGitCoreCommitHash> getOldCommitHash() {
-    return GitCoreCommitHash.ofZeroable(reflogEntry.getOldId());
+    return reflogEntry.getOldId().ofZeroable();
```
Comment (PawelLipski) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreReflogEntry.java:40]: Ditto 🤔

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/GitCommandUpdatingCurrentBranchBackgroundable.java [hunk 4]
```diff
@@ -184,8 +186,8 @@ private void handleResult(
     } else {
       VcsNotifier.getInstance(project).notifyError(
           /* displayId */ null,
-          format(getString("action.GitMachete.GitCommandUpdatingCurrentBranchBackgroundable.notification.title.update-fail"),
-              getOperationName()),
+          getString("action.GitMachete.GitCommandUpdatingCurrentBranchBackgroundable.notification.title.update-fail")
+              .format(getOperationName()),
```
[the remark this answers is not in the document]
Comment (azdrojowa123) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/GitCommandUpdatingCurrentBranchBackgroundable.java:190]: yes :>

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreCommitHash.java [hunk 5]
```diff
@@ -11,12 +11,12 @@ private GitCoreCommitHash(ObjectId objectId) {
...
   }
 
-  public static GitCoreCommitHash of(ObjectId objectId) {
+  public static GitCoreCommitHash toGitCoreCommitHash(ObjectId objectId) {
     return new GitCoreCommitHash(objectId);
   }
 
-  public static Option<IGitCoreCommitHash> ofZeroable(ObjectId objectId) {
-    return objectId.equals(ObjectId.zeroId()) ? Option.none() : Option.some(of(objectId));
+  public static Option<IGitCoreCommitHash> toOptionable(ObjectId objectId) {
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreCommitHash.java:18]: `Optionable` doesn't sound natural

**Review by PawelLipski — COMMENTED** [review]
Also! Vavr static methods to be covered as well, as in the issue

**Comment by azdrojowa123** [comment]
> Also! Vavr static methods to be covered as well, as in the issue
Screenshot 2022-06-01 at 11 03 56

something like in this photo?

**Comment by azdrojowa123** [comment]
> Yeah, `sequence()` FTW; also `map` + `sequence` can be squashed into `traverse`

It a problem here - @ExtensionMethod require static method to be public (must have a `public` modifier), but static methods from e.g. `Try` do not have this modifier

**Comment by PawelLipski** [comment]
> > Yeah, `sequence()` FTW; also `map` + `sequence` can be squashed into `traverse`
> 
> It a problem here - @ExtensionMethod require static method to be public (must have a `public` modifier), but static methods from e.g. `Try` do not have this modifier. And probably more serious problem is that Try is an interface not class

Okay so basically `@ExtensionMethod` doesn't work with static methods in interfaces, right? ☹️ can you see any indication in docs/Lombok issues where does this limitation come from?

**Comment by azdrojowa123** [comment]
> can you see any indication in docs/Lombok issues where does this limitation come from?

I found no explanation from where this limitation comes from. Only in docs is written that 

> You can make a **class** containing a bunch of public, static methods which all take at least 1 parameter

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java [hunk 6]
```diff
@@ -515,7 +516,7 @@ private CreatedAndDuplicatedAndSkippedBranches<NonRootManagedBranchSnapshot> cre
                 "commit (${pointedCommit.getHash().getHashString()}) but parent branch commit " +
                 "is NOT ancestor of parent-agnostic fork point (${parentAgnosticForkPointString}), " +
                 "so we assume that parent-aware fork point = parent branch commit");
-            return ForkPointCommitOfManagedBranch.fallbackToParent(parentPointedCommit);
```
Comment (PawelLipski) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:518]: Huh for `ForkPointCommitOfManagedBranch` I'd leave the original method, it's pretty confusing now

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/FetchAllRemotesAction.java [hunk 7]
```diff
@@ -69,7 +71,7 @@ public void run(ProgressIndicator indicator) {
         result = GitFetchSupport.fetchSupport(project).fetchAllRemotes(gitRepository.toJavaList());
         if (gitRepository.isDefined()) {
           val repoName = gitRepository.get().getRoot().getName();
-          FetchUpToDateTimeoutStatus.update(repoName);
+          repoName.update();
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/FetchAllRemotesAction.java:74]: 👎🏻

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**